### ISSUE TITLE
Problem with RequestOptions "withCredentials: true"

### ISSUE THREAD

**Issue by dspinetti** [oferh/ng2-completer#71]
Hi, I have this code
`this.completerService.urlFormater(
      (term) => { return BaseHttpService.BASE_URL + "/api/subjects?filter=${term}"; }
     );
this.completerService.headers(BaseHttpService.def_headers);`

How can I manage the RequestOptions? I have to specify the withCredentials param as the code below:
`let options = new RequestOptions({ headers: BaseHttpService.def_headers, withCredentials: true });`
Is it possible?
Maybe have sense to expose a RequestOptions attributes like Headers ?

**Comment by oferh** [comment]
Sorry for the late reply.
It can be a good idea to replace headers with `RequestOptions` - will consider that.

At the moment there are two options to do this:

1. If your data is static you can pass the http request observable as `datasource`
```ts
@Component({
  selector: 'my-component',
  template: `<h1>Search color</h1>
            <h1>Search my url</h1>
            <ng2-completer [(ngModel)]="captain" [datasource]="datasource" [minSearchLength]="0"></ng2-completer>`
})
export class MyComponent {
  protected searchStr: string;
  protected datasource: Observable<any[]>;
  constructor(private completerService: CompleterService, private http: Http) {
    let options = new RequestOptions({ withCredentials: true });       
    this.datasource = completerService.local(this.http.get(myUrl, options), 'fieldName', 'fieldName');
  }
}
```
2. create a custom `DataProvider` that will perform the search 
```ts
export class CustomData extends Subject<CompleterItem[]> implements CompleterData {
   constructor(private http: Http) {
      super();
   }
   public search(term: string): void {
       let options = new RequestOptions({ withCredentials: true }); 
       this.http.get(myUrl + term)
            .map((res: Response) => {
                // Convert the result to CompleterItem[]
                let data = res.json();
                this.next(data);
           })
           .subscribe();
   }

 public cancel() {
     // Handle cancel
 }
}
```

**Comment by Tomagi** [comment]
Hi Ofer,
What is  Subject<CompleterItem[]>?

**Comment by oferh** [comment]
[A Subject is a sort of bridge or proxy that is available in some implementations of ReactiveX that acts both as an observer and as an Observable.](http://reactivex.io/documentation/subject.html)

**Comment by oferh** [comment]
closed in 1.4.0
